src/WinRT.Runtime/ActivationFactory.cs

```diff
@@ -0,0 +1,272 @@
+
+#if NET7_0_OR_GREATER || CsWinRT_LANG_11_FEATURES
+            ReadOnlySpan<byte> functionName = "DllGetActivationFactory"u8;
+#else
+            string functionName = "DllGetActivationFactory";
+#endif
+            getActivationFactory = Platform.TryGetProcAddress(moduleHandle, functionName);
+            if (getActivationFactory == null)
+            {
+                module = null;
+                return false;
+            }
+
+            module = new DllModule(
+                fileName,
+                moduleHandle,
+                getActivationFactory);
+            return true;
+        }
+
+        private DllModule(string fileName, IntPtr moduleHandle, void* getActivationFactory)
+        {
+            _fileName = fileName;
+            _moduleHandle = moduleHandle;
+            _GetActivationFactory = (delegate* unmanaged[Stdcall]<IntPtr, IntPtr*, int>)getActivationFactory;
+
+            void* canUnloadNow = null;
+#if NET7_0_OR_GREATER || CsWinRT_LANG_11_FEATURES
+            ReadOnlySpan<byte> functionName = "DllCanUnloadNow"u8;
+#else
+            string functionName = "DllCanUnloadNow";
+#endif
+            canUnloadNow = Platform.TryGetProcAddress(_moduleHandle, functionName);
+
+            if (canUnloadNow != null)
+            {
+                _CanUnloadNow = (delegate* unmanaged[Stdcall]<int>)canUnloadNow;
+            }
+        }
+
+        public unsafe (ObjectReference<IActivationFactoryVftbl> obj, int hr) GetActivationFactory(string runtimeClassId)
+        {
+            IntPtr instancePtr = IntPtr.Zero;
+            try
+            {
+                MarshalString.Pinnable __runtimeClassId = new(runtimeClassId);
+                fixed (void* ___runtimeClassId = __runtimeClassId)
+                {
+                    int hr = _GetActivationFactory(MarshalString.GetAbi(ref __runtimeClassId), &instancePtr);
+                    if (hr == 0)
+                    {
+                        var objRef = ObjectReference<IActivationFactoryVftbl>.Attach(ref instancePtr);
+                        return (objRef, hr);
+                    }
+                    else
+                    {
+                        return (null, hr);
+                    }
+                }
+            }
+            finally
+            {
+                MarshalInspectable<object>.DisposeAbi(instancePtr);
+            }
+        }
+
+        ~DllModule()
+        {
+            System.Diagnostics.Debug.Assert(_CanUnloadNow == null || _CanUnloadNow() == 0); // S_OK
+            lock (_cache)
+            {
+                _cache.Remove(_fileName);
+            }
+            if ((_moduleHandle != IntPtr.Zero) && !Platform.FreeLibrary(_moduleHandle))
+            {
+                Marshal.ThrowExceptionForHR(Marshal.GetHRForLastWin32Error());
+            }
+        }
+    }
+
+    internal sealed class WinrtModule
+    {
+        readonly IntPtr _mtaCookie;
+        volatile static WinrtModule _instance;

```

Comment:
Nit: [incorrect modifiers order](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0036), should be `static volatile`. Also, actually, add `private` as well (same in the other fields).